### PR TITLE
Fix dynamic plugin import

### PR DESCRIPTION
**Pull request by zAlweNy26**
# Description

With this PR I fixed the import for the plugins that was causing a bug.

Related to issue #514 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: core/cat/mad_hatter/plugin.py [hunk 1]
```diff
@@ -207,18 +207,18 @@ def _load_decorated_functions(self):
...
-            py_filename = py_file.replace("/", ".").replace(".py", "")  # this is UGLY I know. I'm sorry
+            module_name = os.path.splitext(os.path.basename(py_file))[0]
 
-            log.info(f"Import module {py_filename}")
+            log.info(f"Import module {py_file}")
 
             # save a reference to decorated functions
             try:
-                plugin_module = importlib.import_module(py_filename)
+                plugin_module = imp.load_source(module_name, py_file)
```
Comment (nicola-corbellini) [inline core/cat/mad_hatter/plugin.py:216]: why not importlib?

Comment (zAlweNy26) [reply core/cat/mad_hatter/plugin.py:216]: I didn't find a clean way to do it with importlib. If you have a better way to do that, I accept suggestions 👍🏽

Comment (nicola-corbellini) [reply core/cat/mad_hatter/plugin.py:216]: Can you try something like this, please?
```python
spec=importlib.util.spec_from_file_location(module_name, py_file)
plugin_module = importlib.util.module_from_spec(spec)
```
I think `imp` package is deprecated

Comment (zAlweNy26) [reply core/cat/mad_hatter/plugin.py:216]: I tried that way but it says it doesn't find any hooks

Comment (zAlweNy26) [reply core/cat/mad_hatter/plugin.py]: @nicola-corbellini I found a way to use importlib, if you want to check it out

File: core/cat/mad_hatter/plugin.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ def _load_decorated_functions(self):
 
             # save a reference to decorated functions
             try:
-                plugin_module = imp.load_source(module_name, py_file)
+                plugin_module = machinery.SourceFileLoader(module_name, py_file).load_module()
```
Comment (nicola-corbellini) [inline core/cat/mad_hatter/plugin.py:216]: Top level Py-dev

Comment (zAlweNy26) [reply core/cat/mad_hatter/plugin.py:216]: Are you trolling me?

Comment (nicola-corbellini) [reply core/cat/mad_hatter/plugin.py:216]: I'm greeting you

**Pull request merged**